### DISCUSSION TITLE
POST Requests Aren't Sent Unless Await Is Used

### DISCUSSION THREAD

**Discussion by AlexIsMaking** [vercel/community#2669] [Help] [answered]
### Summary

Pretty simple - I'm sending an API request from one Vercel project to another. If I use

```
  await axios.post(upsertPage, {
    var1,
    var2,
    var3,
  });
```

then the request is received but if I remove the `await` then it isn't.

### Steps to Reproduce

See above.

**Comment by smaeda-ks** [comment, marked as answer]
@AlexIsMaking That is expected, and you should always use `await` properly.

- https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html
- https://levelup.gitconnected.com/avoiding-the-pitfalls-of-async-node-js-functions-in-aws-lambda-941220582e7a

**Reply by AlexIsMaking** [reply]
Thanks for letting me know. That's unhelpful because I want to speed up the execution of that step in my code and don't mind what the result of the API request is but if it's a limitation of a service that Vercel's using then fair enough.

**Reply by smaeda-ks** [reply]
That's only applied to Serverless Functions. For Edge Functions, you could use `waitUntil()`:
https://vercel.com/docs/concepts/functions/edge-functions/edge-functions-api#waituntil

**Reply by AlexIsMaking** [reply]
Thanks for sharing this, I'll probably need to use Node.js APIs in the project that I'm working on (it'd be nice if there was an easy way to check whether any of the packages that you're using uses them) but I expect that will be useful in the future.
